src/components/application_manager/test/commands/mobile/show_constant_tbt_test.cc

```diff
@@ -0,0 +1,402 @@
+  (*message_)[am::strings::msg_params][am::strings::navigation_text_1] =
+      kNavigationText1 + "\n";
+
+  ExpectedManageMobileCommandResultCode(mobile_apis::Result::INVALID_DATA);
+
+  request_ = CreateCommand<ShowConstantTBTRequest>(message_);
+  ASSERT_TRUE(request_->Init());
+  request_->Run();
+}
+
+TEST_F(ShowConstantTBTRequestTest,
+       IsWhiteSpaceExist_NavigationText2InvalidSyntax_InvalidData) {
+  (*message_)[am::strings::msg_params][am::strings::navigation_text_2] =
+      kNavigationText2 + "\n";
+
+  ExpectedManageMobileCommandResultCode(mobile_apis::Result::INVALID_DATA);
+
+  request_ = CreateCommand<ShowConstantTBTRequest>(message_);
+  ASSERT_TRUE(request_->Init());
+  request_->Run();
+}
+
+TEST_F(ShowConstantTBTRequestTest,
+       IsWhiteSpaceExist_EtaInvalidSyntax_InvalidData) {
+  (*message_)[am::strings::msg_params][am::strings::eta] = kEta + "\n";
+
+  ExpectedManageMobileCommandResultCode(mobile_apis::Result::INVALID_DATA);
+
+  request_ = CreateCommand<ShowConstantTBTRequest>(message_);
+  ASSERT_TRUE(request_->Init());
+  request_->Run();
+}
+
+TEST_F(ShowConstantTBTRequestTest,
+       IsWhiteSpaceExist_TotalDistanceInvalidSyntax_InvalidData) {
+  (*message_)[am::strings::msg_params][am::strings::total_distance] =
+      kTotalDistance + "\n";
+
+  ExpectedManageMobileCommandResultCode(mobile_apis::Result::INVALID_DATA);
+
+  request_ = CreateCommand<ShowConstantTBTRequest>(message_);
+  ASSERT_TRUE(request_->Init());
+  request_->Run();
+}
+
+TEST_F(ShowConstantTBTRequestTest,
+       IsWhiteSpaceExist_TimeToDestinationInvalidSyntax_InvalidData) {
+  (*message_)[am::strings::msg_params][am::strings::total_distance] =
+      kTotalDistance + "\n";
+
+  ExpectedManageMobileCommandResultCode(mobile_apis::Result::INVALID_DATA);
+
+  request_ = CreateCommand<ShowConstantTBTRequest>(message_);
+  ASSERT_TRUE(request_->Init());
+  request_->Run();
+}
+
+TEST_F(ShowConstantTBTRequestTest,
+       OnEvent_NavigationShowConstantTBT_SendMobileResponse) {
+  MessageSharedPtr message = CreateMessage(smart_objects::SmartType_Map);
+  (*message)[am::strings::params][am::hmi_response::code] =
+      hmi_apis::Common_Result::SUCCESS;
+  (*message)[am::strings::msg_params][am::strings::info] = "info1";
+
+  Event event(hmi_apis::FunctionID::Navigation_ShowConstantTBT);
+  event.set_smart_object(*message);
+
+  ExpectedManageMobileCommandResultCode(mobile_apis::Result::SUCCESS);
+
+  request_ = CreateCommand<ShowConstantTBTRequest>(message_);
+  request_->on_event(event);
+}
+
+TEST_F(ShowConstantTBTRequestTest, OnEvent_InvalidEnum_ReceivedUnknownEvent) {
+  MessageSharedPtr message = CreateMessage(smart_objects::SmartType_Map);
+
+  Event event(hmi_apis::FunctionID::INVALID_ENUM);
+  event.set_smart_object(*message);
+
+  EXPECT_CALL(app_mngr_, ManageMobileCommand(_, _)).Times(0);
+
+  request_ = CreateCommand<ShowConstantTBTRequest>(message_);
+  request_->on_event(event);
```

Comment:
@BSolonenko Every on_event method depends on Init & Run. Call the before every on_event.